lib/Canto/UniProt/UniProtUtil.pm

```diff
@@ -98,6 +98,17 @@ sub parse_results
         }
       }
 
+      if (!defined $name) {
+        for my $synonym (@{$entry->{gene}->[0]->{name}}) {
+          my $synonym_content = _content($synonym);
+
+          if ($synonym->{type} eq 'ordered locus') {
```

Comment:
Could this check be merged with the check for ORF names below? It seems like the other logic and iteration is redundant. Here's what I mean:

```pl
if ($synonym->{type} eq 'ordered locus' ||  $synonym->{type} eq 'ORF') {
```
The short-circuiting should mean that ordered locus names have higher precedence than ORF names. I've tested this on an accession with both name types ([Q0DHH6](https://www.uniprot.org/uniprotkb/Q0DHH6/entry)) and it uses the ordered locus name (although I did have to refresh the gene cache first for some reason). I also tested an accession with only an ORF name ([A0A1Q4I1Y1](https://www.uniprot.org/uniprotkb/A0A1Q4I1Y1/entry)) and that uses the ORF name instead.

Reply:
I wanted to make sure the "ordered locus" names where given priority if there are cases like this:

```xml
    <gene>
      <name type="primary">POL3</name>
      <name type="ORF">D2366</name>
      <name type="ordered locus">YDL102W</name>
    </gene>
```

If we have just one loop that tests for "ordered locus" and "ORF", for this XML the ORF name will be found first and stored in `$name`.  The  "ordered locus" won't be reached.